validator/client/validator_attest.go

```diff
@@ -112,6 +112,22 @@ func (v *validator) SubmitAttestation(ctx context.Context, slot uint64, pubKey [
 			return
 		}
 	}
+	if featureconfig.Get().SlasherProtection && v.protector != nil {
```

Comment:
`SubmitAttestation` has gotten pretty messy now with 2 slashing protection routines... the first one is above:

`if featureconfig.Get().ProtectAttester {....`

And now there's:

`if featureconfig.Get().SlasherProtection && v.protector != nil {`....

How can we make this better?

Reply:
Maybe we can make a "preAttestationCreation" function or something that holds this kind of functionality. Generalize it a bit more instead of having it in the main function?

Reply:
lgtm

Reply:
sg @terencechain WDYT?

Reply:
doing it now